src/files.ts

```diff
@@ -431,7 +431,8 @@ export function resolveBundleFilePath(baseDir: string, bundleFilePath: string):
   return decodeURI(relPath);
 }
 
-export function* composeFilePayloads(files: FileInfo[], bucketSize = MAX_PAYLOAD): Generator<FileInfo[]> {
+// MAX_PAYLOAD / 2 is because every char takes 2 bytes in the payload
+export function* composeFilePayloads(files: FileInfo[], bucketSize = MAX_PAYLOAD / 2): Generator<FileInfo[]> {
```

Comment:
q: Do the changes in this function have an impact on production functionality?

Reply:
only on client. this function is responsible of creating chunks of data, that can fit in one request. As a result of these changes, it will actually create smaller chunks. But won't have an impact. Code-client is used now only in CLI and old version of VSCe. Both will work well. Potentially, it can cause slower rate of download of initial project files. But based on my load tests, adjustments of payload size give max up to 10% difference in duration. In this can, it will be even less, thus unnoticeable

Reply:
I see, thanks for the explanation. So what's the motivation for this change then if it causes a (slightly) worse UX?

Rest of the PR LGTM btw

Reply:
This change makes the calculation more correct and predictable. Less likely to extend beyond allowed limits. So, i believe more accurate calculation is worth unnoticeably worsen performance.